Code/src/main/java/nl/utwente/group10/ui/components/anchors/InputAnchor.java

```diff
@@ -1,42 +1,46 @@
 package nl.utwente.group10.ui.components.anchors;
 
-import java.util.Optional;
-
-import edu.emory.mathcs.backport.java.util.Arrays;
 import nl.utwente.group10.haskell.expr.Expr;
 import nl.utwente.group10.haskell.expr.Ident;
-import nl.utwente.group10.haskell.type.Type;
-import nl.utwente.group10.ui.CustomUIPane;
 import nl.utwente.group10.ui.components.blocks.Block;
-import nl.utwente.group10.ui.components.blocks.InputBlock;
 import nl.utwente.group10.ui.components.lines.Connection;
-import nl.utwente.group10.ui.exceptions.TypeUnavailableException;
 import nl.utwente.group10.ui.handlers.AnchorHandler;
 
 /**
- * Anchor that specifically functions as an input.
+ * ConnectionAnchor that specifically functions as an input.
  */
 public class InputAnchor extends ConnectionAnchor {
+    /** The expression to return when there is no connection. */
+    private Expr connectionlessExpr;
```

Comment:
Are you sure this is correct? If you store the Expr (and its Type), I assume the unified type will stick around, yes? Why not always respond with a new `undefined`?